custom_components/foxess_modbus/entities/modbus_select.py

```diff
@@ -64,7 +64,13 @@ def current_option(self) -> str | None:
             return None
         if not self._validate(self.entity_description.validate, value):
             return None
-        return self.entity_description.options_map.get(value)
+
+        options = self.entity_description.options_map
+        if value not in options:
+            _LOGGER.warning(
+                "Value (%s) is not valid. Valid values: (%s)", value, self.options
+            )
+        return options.get(value)
```

Comment:
Nit: we can avoid the double-lookup by doing `get` once, then checking whether we got `None`.

Also log the entity id / address maybe?